netket/operator/_hamiltonian.py

```diff
@@ -27,46 +28,40 @@ def conjugate(self, *, concrete: bool = True):
     def __add__(self, other):
         if type(self) is type(other):
             res = self.copy()
-            res += other
-            return res
-
-        return self.to_local_operator().__add__(other)
+            res = res.__iadd__(other)
+            if res is not NotImplemented:
```

Comment:
ah wait what? is not NotImplemented? can we simplify the logic ?

Reply:
I changed it to `isinstance(res, SpecialHamiltonian)`. Get's rid of the double negation but it's less clear what the check is for.